lib/ethpy/tests/test_interface.py

```diff
@@ -0,0 +1,119 @@
+        "baseToken": hyperdrive_contract_addresses.base_token,
+        "initialSharePrice": FixedPoint("1"),
+        "minimumShareReserves": FixedPoint("10"),
+        "minimumTransactionAmount": FixedPoint("0.001"),
+        "positionDuration": 604800,  # 1 week
+        "checkpointDuration": 3600,  # 1 hour
+        "timeStretch": expected_timestretch_fp,
+        "governance": deploy_account.address,
+        "feeCollector": deploy_account.address,
+        "curveFee": FixedPoint("0.1"),  # 10,
+        "flatFee": FixedPoint("0.0005"),  # 0.0%
+        "governanceFee": FixedPoint("0.15"),  # 1%
+        "oracleSize": 10,
+        "updateGap": 3600,  # TODO don't know where this is getting set
+        "invTimeStretch": (1 / expected_timestretch_fp),
+    }
+    expected_pool_config["fees"] = [
+        expected_pool_config["curveFee"],
+        expected_pool_config["flatFee"],
+        expected_pool_config["governanceFee"],
+    ]
+
+    api_pool_config = interface.pool_config
+
+    # Existence test
+    assert len(api_pool_config) > 0, "API pool config must have length greater than 0"
+
+    # Ensure keys match
+    # Converting to sets and compare
+    api_keys = set(api_pool_config.keys())
+    expected_keys = set(expected_pool_config.keys())
+    assert api_keys == expected_keys, "Keys in API do not match expected"
+
+    # Value comparison
+    for key, expected_value in expected_pool_config.items():
+        print(f"{expected_value=}")
+        print(f"{api_pool_config[key]=}")
+        assert_val = api_pool_config[key] == expected_value
+        assert assert_val, f"Values do not match for {key} ({api_pool_config[key]} != {expected_value})"
+
+    # Pool info comparison
+    api_pool_info = interface.pool_info
+    expected_pool_info_keys = [
+        # Keys from contract call
+        "blockNumber",
+        "shareReserves",
+        "bondReserves",
+        "lpTotalSupply",
+        "sharePrice",
+        "shareAdjustment",
+        "lpSharePrice",
+        "longExposure",
+        "longsOutstanding",
+        "longAverageMaturityTime",
+        "shortsOutstanding",
+        "shortAverageMaturityTime",
+        "withdrawalSharesReadyToWithdraw",
+        "withdrawalSharesProceeds",
+        # Added keys
+        "timestamp",
+        # Calculated keys
+        "totalSupplyWithdrawalShares",
+    ]
+    # Convert to sets and compare
+    assert set(api_pool_info.keys()) == set(expected_pool_info_keys)
+
+    # Check spot price and fixed rate
+    api_spot_price = interface.spot_price
+    expected_spot_price = calc_spot_price(
+        share_reserves=api_pool_info["shareReserves"],
+        share_adjustment=api_pool_info["shareAdjustment"],
+        bond_reserves=api_pool_info["bondReserves"],
+        initial_share_price=api_pool_config["initialSharePrice"],
+        time_stretch=api_pool_config["timeStretch"],
+    )
+
+    api_fixed_rate = interface.fixed_rate
+    expected_fixed_rate = calc_fixed_rate_fp(
```

Comment:
I'm hesitant to depend on a function in analysis in a test, I suspect this function will get replaced by the interface at some point.

Reply:
fixed in c906830f59f6d8e8d35d417b38818d4b431e9e5f